crates/bevy_utils/src/float_ord.rs

```diff
@@ -64,3 +84,37 @@ impl Neg for FloatOrd {
         FloatOrd(-self.0)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn float_ord_eq() {
+        let nan = FloatOrd(f32::NAN);
+        let num = FloatOrd(0.0);
+
+        assert_eq!(nan, nan);
+
+        assert_ne!(nan, num);
+        assert_ne!(num, nan);
+
+        assert_eq!(num, num);
+    }
+
+    #[test]
+    fn float_ord_cmp() {
+        let nan = FloatOrd(f32::NAN);
+        let zero = FloatOrd(0.0);
+        let one = FloatOrd(1.0);
+
+        assert_eq!(nan.cmp(&nan), Ordering::Equal);
+
+        assert_eq!(nan.cmp(&zero), Ordering::Less);
+        assert_eq!(zero.cmp(&nan), Ordering::Greater);
+
+        assert_eq!(zero.cmp(&zero), Ordering::Equal);
+        assert_eq!(one.cmp(&zero), Ordering::Greater);
+        assert_eq!(zero.cmp(&one), Ordering::Less);
+    }
+}
```

Comment:
could we add some tests for `le` too, since the behavior there is a little weird and should match `cmp`?

Reply:
Done for all operators, also added a test for the `Hash` impl.